std-bits/table/src/main/java/org/enso/table/write/ExcelWriter.java

```diff
@@ -0,0 +1,323 @@
+        Storage storage = storages[j];
+        int idx = j + firstColumn - 1;
+
+        Cell cell = row.getCell(idx);
+        if (cell == null) {
+          cell = row.createCell(idx);
+        }
+
+        writeValueToCell(cell, i, storage, workbook);
+      }
+      currentRow++;
+    }
+
+    workbook.setForceFormulaRecalculation(true);
+  }
+
+  private static CellStyle getDateTimeStyle(Workbook workbook, String format) {
+    for(int i = 0; i < workbook.getNumCellStyles(); i++) {
+      CellStyle style = workbook.getCellStyleAt(i);
+      if (style.getDataFormatString().equals(format)) {
+        return style;
+      }
+    }
+
+    CellStyle newStyle = workbook.createCellStyle();
+    newStyle.setDataFormat(workbook.createDataFormat().getFormat(format));
+    return newStyle;
+  }
+
+  private static void writeValueToCell(Cell cell, int j, Storage storage, Workbook workbook)
+    throws IllegalStateException {
+    if (storage.isNa(j)) {
+      cell.setBlank();
+    } else if (storage instanceof DoubleStorage doubleStorage) {
+      cell.setCellValue(doubleStorage.getItem(j));
+    } else if (storage instanceof LongStorage longStorage) {
+      cell.setCellValue(longStorage.getItem(j));
+    } else if (storage instanceof BoolStorage boolStorage) {
+      cell.setCellValue(boolStorage.getItem(j));
+    } else {
+      Object value = storage.getItemBoxed(j);
+      switch (value) {
+        case String s -> cell.setCellValue(s);
+        case Boolean b -> cell.setCellValue(b);
+        case Double d -> cell.setCellValue(d);
+        case Long l -> cell.setCellValue(l);
+        case LocalDateTime ldt -> {
+          cell.setCellValue(ldt);
+          cell.setCellStyle(getDateTimeStyle(workbook, "yyyy-MM-dd HH:mm:ss"));
+        }
+        case LocalDate ld -> {
+          cell.setCellValue(ld);
+          cell.setCellStyle(getDateTimeStyle(workbook, "yyyy-MM-dd"));
+        }
+        case LocalTime lt -> {
+          cell.setCellValue(lt.toSecondOfDay() / SECONDS_IN_A_DAY);
+          cell.setCellStyle(getDateTimeStyle(workbook, "HH:mm:ss"));
+        }
+        default -> {
+          if (ensoToTextCallback != null) {
+            cell.setCellValue(ensoToTextCallback.apply(value));
+          } else {
+            throw new IllegalStateException("Enso to text callback is not set. Unable to process value.");
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * Determines if an existing range has headers.
+   * Unlike in the ExcelReader, if empty this will default to True.
+   * @param excelSheet the Excel sheet to check.
+   * @param topRow top row index (1-based) of the range to check.
+   * @param startCol start column index (1-based) of the range to check.
+   * @param endCol end column index (1-based) of the range to check. If -1 will continue until end of row.
+   * @return true if the range has headers.
+   */
+  private static boolean shouldWriteHeaders(ExcelSheet excelSheet, int topRow, int startCol, int endCol) {
```

Comment:
I guess now the comment should be updated too?

```suggestion
  /**
   * Determines if headers should be written for the given range in {@code INFER} mode.
   *
   * Unlike in the {@code ExcelReader}, if empty this will default to True.
   * @param excelSheet the Excel sheet to check.
   * @param topRow top row index (1-based) of the range to check.
   * @param startCol start column index (1-based) of the range to check.
   * @param endCol end column index (1-based) of the range to check. If -1 will continue until end of row.
   * @return true if the range has headers.
   */
  private static boolean shouldWriteHeaders(ExcelSheet excelSheet, int topRow, int startCol, int endCol) {
```